scripts/prepare

```diff
@@ -61,5 +61,16 @@ for f in packages/*; do
 				rm -rf $dep
 			done
 		fi
-  fi
+	fi
+done
+
+# if any sub-charts were prepared pull them in
+for f in packages/*; do
+	if [[ ! -f ${f}/package.yaml && -f ${f}/charts/requirements.yaml ]]; then
+		ls ${f}/charts/requirements.yaml
+		pwd=$(pwd)
+		cd ${f}/charts
+		helm dependency update
+		cd $pwd
+	fi
 done
```

Comment:
@StrongMonkey can you review this idea ^. Basically I am packaging a chart (kiali) separately in charts and now need to call `helm dependency update` in a non upstream chart to pull it in after it gets generated.

Reply:
Looks good to me.

Reply:
@cbron I believe we should just move the following lines of code outside the `if [[ -z $CHART || $CHART == $(basename -- ${f}) ]]; then` statement:

https://github.com/rancher/charts/blob/f6f0c3ddbe3c56e62f3b9c32b4964a1c5fe230f5/scripts/prepare#L42-L62

Reply:
The `copied_dependencies` logic will store nothing for charts without `overlay/` directories (i.e. for non upstream charts) since it checks to see if the overlay chart exists before doing the `cp -R` or adding it to the list of `copied_dependencies`. 

So the code path should be the same regardless of a `package.yaml` exists.

Reply:
One more change will also be required: we need to flip these two if statements

https://github.com/rancher/charts/blob/f6f0c3ddbe3c56e62f3b9c32b4964a1c5fe230f5/scripts/prepare#L8-L9

The logic I mentioned above will be within the`if [[ -z $CHART || $CHART == $(basename -- ${f}) ]]; then` statement but outside the `if [[ -f ${f}/package.yaml ]]; then` statement. This will avoid running the `helm dependency update` on all charts when a specific `CHART` is selected via env var.

Reply:
The problem with that approach is that the generated charts need to exist before they can be vendored in via `helm dependency`. So if we loop the charts and kiali comes after istio, then kiali won't exist when istio tries to vendor. 

But on 3rd part yes I need to wrap in `$CHART` check.

Reply:
added that check

Reply:
Good catch; then I agree we should use two for loops, but we can just move https://github.com/rancher/charts/blob/f6f0c3ddbe3c56e62f3b9c32b4964a1c5fe230f5/scripts/prepare#L42-L62 into the second for loop instead of having the same logic twice.

Neither for loop here should impact charts that do not use package.yaml since such charts will not have an `overlay/` folder, which means no `copied_dependencies` will ever be added due to this: https://github.com/rancher/charts/blob/f6f0c3ddbe3c56e62f3b9c32b4964a1c5fe230f5/scripts/prepare#L49

Reply:
updated, @StrongMonkey want to have a look again